app/controllers/application_controller.rb

```diff
@@ -16,6 +16,19 @@ class ApplicationController < ActionController::Base
 
   protected
 
+  def current_admin_user
+    return current_user if current_user && current_user.role == "admin"
```

Comment:
you can use `current_user.admin?` here if you add to the enumerize declaration `predicates: true`. Also we need to return nil not false as this is not a question method.